### PR TITLE
mail-client/claws-mail-4.1.1: add oauth flag

### PR DESCRIPTION
**Pull request by kuraga**
Claws Mail 4.1.1 made OAuth2 support optional: https://git.claws-mail.org/?p=claws.git;a=commit;h=d1d9ffd6c37e25c6e25bfb59571591824122b91e

Closes: https://bugs.gentoo.org/878305

### CODE REVIEW

File: mail-client/claws-mail/claws-mail-4.1.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+else
+	SRC_URI="https://www.claws-mail.org/download.php?file=releases/${P}.tar.xz"
+	KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ppc ~ppc64 ~riscv ~sparc ~x86"
+fi
+
+SLOT="0"
+QA_PKGCONFIG_VERSION="${PV}.0"
+LICENSE="GPL-3"
+
+IUSE="archive bogofilter calendar clamav dbus debug dillo doc gdata +gnutls +oauth +imap ldap +libcanberra +libnotify litehtml networkmanager nls nntp +notification pdf perl +pgp rss session sieve smime spamassassin spam-report spell startup-notification svg valgrind webkit xface"
```
Comment (voyageur) [inline mail-client/claws-mail/claws-mail-4.1.1-r1.ebuild:25]: Let's keep the USE flags sorted (especially here with all the possible ones) - "+oauth" should be added just before "pdf"

File: mail-client/claws-mail/claws-mail-4.1.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+REQUIRED_USE="
+	libcanberra? ( notification )
+	libnotify? ( notification )
+	networkmanager? ( dbus )
+	smime? ( pgp )
+	oauth? ( gnutls )
```
Comment (voyageur) [inline mail-client/claws-mail/claws-mail-4.1.1-r1.ebuild:31]: Less important than in USE flags but keeping this list sorted is helpful in maintenance

File: mail-client/claws-mail/claws-mail-4.1.1-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+		$(use_enable bogofilter bogofilter-plugin)
+		$(use_enable calendar vcalendar-plugin)
+		$(use_enable clamav clamd-plugin)
+		$(use_enable dbus)
+		$(use_enable debug crash-dialog)
+		$(use_enable dillo dillo-plugin)
+		$(use_enable doc manual)
+		$(use_enable gdata gdata-plugin)
+		$(use_enable gnutls)
+		$(use_enable oauth oauth2)
```
Comment (voyageur) [inline mail-client/claws-mail/claws-mail-4.1.1-r1.ebuild:149]: And same in configure flags :)

**Review by voyageur — CHANGES_REQUESTED** [review]
Thanks again for the PR and bug! Aside from placing the new USE flag in sorted order, we will also need an addition to metadata.xml
oauth is a [local USE flag](https://devmanual.gentoo.org/ebuild-writing/misc-files/metadata/index.html#local-use-flag-descriptions) so it needs an entry there, I think something like the one used for gnome-extra/evolution-data-server will be fine: "Enable internal OAuth2 authentication support"

You can spot this issue (and similar) running "pkgcheck scan" from dev-util/pkgcheck

**Comment by kuraga** [comment]
> we will also need an addition to metadata.xml oauth is a [local USE flag](https://devmanual.gentoo.org/ebuild-writing/misc-files/metadata/index.html#local-use-flag-descriptions) so it needs an entry there

Done.

P.S. Hm... I didn't see `local` [here](https://packages.gentoo.org/useflags/oauth). Is it strange to take description from the first package used this USE flag? :)

**Comment by voyageur** [comment]
Indeed, you have to hope the first alphabetical package has a "generic" description - but having a better system would probably require some magic to parse all strings etc :)

**Pull request closed without merging**